libflux/src/core/ast/mod.rs

```diff
@@ -542,8 +542,18 @@ pub struct TypeExpression {
     #[serde(flatten)]
     pub base: BaseNode,
     pub monotype: MonoType,
+    pub constraints: Option<Constraints>,
 }
 
+#[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
+pub struct Constraints {
```

Comment:
Should we call this `TypeConstraints`?

Reply:
I think we can make it ConstraintType to match the others?